### PR TITLE
[QSP-3] feat: add add_ssn2

### PR DESCRIPTION
**Pull request by renlulu**
This pr is to address #25

### CODE REVIEW

File: contracts/ssnlist.scilla [hunk 1]
```diff
@@ -300,13 +300,40 @@ end
...
+(* @dev: Only used for contract upgrade *)
+(* @param ssnaddr: Address of the ssn to be added *)
 (* @param stake_amount: Stake amount of ssn *)
 (* @param rewards: Rewards of the ssn to be added *)
 (* @param urlraw: string representing "ip:port" of the ssn serving raw api request *)
 (* @param urlapi: string representing url exposed by ssn serving public api request *)
 (* @param buffered_deposit: Any buffered stake deposit *)
 (* @param initiator: The original caller who called the proxy *)
-transition add_ssn (ssnaddr : ByStr20, stake_amount : Uint128, rewards : Uint128, urlraw : String, urlapi : String, buffered_deposit : Uint128, initiator : ByStr20)
+transition add_ssn2 (ssnaddr : ByStr20, stake_amount : Uint128, rewards : Uint128, urlraw : String, urlapi : String, buffered_deposit : Uint128, initiator : ByStr20)
```
Comment (bb111189) [inline contracts/ssnlist.scilla:336]: rename `add_ssn2` to `add_ssn_after_upgrade`

File: contracts/proxy.scilla [hunk 2]
```diff
@@ -120,22 +120,23 @@ transition update_contractmaxstake (max_stake : Uint128)
...
-
 (* @dev: Adds new ssn to ssnlist. Used by admin only. *)
 (* @param ssnaddr: Address of the ssn to be added *)
 (* @param stake_amount: Stake amount of ssn *)
 (* @param rewards: Rewards of the ssn to be added *)
 (* @param urlraw: string representing "ip:port" of the ssn serving raw api request *)
 (* @param urlapi: string representing url exposed by ssn serving public api request *)
 (* @param buffered_deposit: Any buffered stake deposit *)
-transition add_ssn (ssnaddr : ByStr20, stake_amount : Uint128, rewards : Uint128, urlraw : String, urlapi : String, buffered_deposit : Uint128)
+transition add_ssn (ssnaddr : ByStr20, rewards : Uint128, urlraw : String, urlapi : String, buffered_deposit : Uint128)
```
Comment (bb111189) [inline contracts/proxy.scilla:130]: bufffered_deposit should be zero right?

Comment (bb111189) [reply contracts/proxy.scilla:130]: reward should be 0 too

Comment (renlulu) [reply contracts/proxy.scilla:130]: Yes, I will remove `buffered_deposit ` too. Thanks.

File: contracts/ssnlist.scilla [hunk 3]
```diff
@@ -300,13 +300,40 @@ end
 
 (* @dev: Adds new ssn to ssnlist. Used by admin only. *)
 (* @param ssnaddr: Address of the ssn to be added *)
+(* @param rewards: Rewards of the ssn to be added *)
+(* @param urlraw: string representing "ip:port" of the ssn serving raw api request *)
+(* @param urlapi: string representing url exposed by ssn serving public api request *)
+(* @param initiator: The original caller who called the proxy *)
+transition add_ssn (ssnaddr : ByStr20, rewards : Uint128, urlraw : String, urlapi : String, initiator : ByStr20)
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:307]: Should we remove `rewards` as a param and keep it to zero like other fields for the SSN?

Comment (bb111189) [reply contracts/ssnlist.scilla:307]: I think we can remove it

Comment (renlulu) [reply contracts/ssnlist.scilla:307]: Yea, will do it, thanks!

**Pull request merged**